db/seeds/01-users.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+const bcrypt = require('bcryptjs')
+const { randomUUID } = require('crypto')
+
+const DatabaseConfig = require('../../config/database.config.js')
```

Comment:
I think (as in you have forced me to think why I did this!) when we are talking about a value as opposed to a function it's best to leave the object so you have some context when it appears in the code, for example, I'm using it in

```javascript
function _generateHashedPassword () {
  const salt = bcrypt.genSaltSync(10)

  return bcrypt.hashSync(DatabaseConfig.defaultUserPassword, salt)
}
```

Remove `DatabaseConfig` it starts looking a bit like it appears from nowhere.

```javascript
function _generateHashedPassword () {
  const salt = bcrypt.genSaltSync(10)

  return bcrypt.hashSync(defaultUserPassword, salt)
}
```

I am conscious they are both just objects declared at the top of the file.

Again, I'm not precious about it either 🤷 . But it feels like a 'convention-in-the-making'. So, one to think about and discuss on the Dev call in my absence!

Reply:
As we are having a bit of a discussion about this :grin: I was wondering why we are generating the `salt` separately? I think we get the same result if we just do this `return bcrypt.hashSync(defaultUserPassword, 10)`.

Reply:
Just following orders 🤷 https://github.com/kelektiv/node.bcrypt.js#to-hash-a-password-1

Reply:
Have a look at **Technique 2** in that readme :wink:

I would guess that the technique to generate the `salt` separately would be more efficient if you were performing multiple hashes as you could reuse the `salt` to save a bit of CPU? Not fussed either way but thought it mention it since we were having a chat about that chunk of code.

Reply:
That's a valid point. I didn't dig that far because I was just trying to replicate what they had done so we generate passwords the rest of the service can recognise. But I think technique 2 makes the code just a bit simpler so leave this with me and I'll make the change.